connector/connector.py

```diff
@@ -227,6 +231,33 @@ def get_binder_for_model(self, model=None):
         log_deprecate('renamed to binder_for()')
         return self.binder_for(model=model)
 
+    @contextmanager
+    def try_advisory_lock(self, lock, retry_seconds=1):
+        """ Context manager, tries to acquire a Postgres transactional
+        advisory lock.
+
+        If the lock cannot be acquired, it raises a
+        ``RetryableJobError`` so the jobs is retried after n
+        ``retry_seconds``.
+
+        See :func:``openerp.addons.connector.connector.pg_try_advisory_lock``
+        for details.
+
+        :param lock: The lock name. Can be anything convertible to a
+           string.  It needs to represents what should not be synchronized
+           concurrently so usually the string will contain at least: the
+           action, the backend type, the backend id, the model name, the
+           external id
+        :param retry_seconds: number of seconds after which a job should
+           be retried when the lock cannot be acquired.
+        """
+        if pg_try_advisory_lock(self.env, lock):
+            yield
```

Comment:
Yeah, it can be clearer, but it isn't also very confused this way. What you prefer.